numpy/core/include/numpy/npy_common.h

```diff
@@ -1030,7 +1030,7 @@ typedef struct { npy_longdouble real, imag; } npy_clongdouble;
 
 /* half/float16 isn't a floating-point type in C */
 #define NPY_FLOAT16 NPY_HALF
-typedef npy_uint16 npy_half;
+typedef struct npy_half { npy_uint16 value;} npy_half;
```

Comment:
Just a thought, but should we call this `bits` maybe?  `value` seem to point at the exact opposite ;).